core/templates/pages/preferences-page/preferences-page.component.ts

```diff
@@ -87,6 +91,24 @@ angular.module('oppia').component('preferencesPage', {
         _saveDataItem('user_bio', userBio);
       };
 
+      ctrl.checkBioChanged = function() {
+        ctrl.allowreload = false;
+      };
+
+      ctrl.saveBeforeUnload = function() {
+        WindowRef.nativeWindow.addEventListener(
+          'beforeunload', ctrl.confirmBeforeLeaving);
+      };
+
+      ctrl.confirmBeforeLeaving = function(e) {
+        if (!ctrl.allowreload) {
+          // This message is irrelevant, but is needed to trigger the
+          // confirmation before leaving.
```

Comment:
Do we need to call `preventDefault` here?  (as per the standard [[ref](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event)])

Reply:
Activation using `event.preventDefault()` does not work in chrome, edge etc. So we use Activation using `event.returnValue = "string";` instead

Reply:
For the legacy purpose, the standard is to use both (preventDefault and returnValue).

Reply:
[Link](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event#browser_compatibility) for compatibility

Reply:
As per offline discussion, we found that the ref doc suggests having both the preventDefault and returnValue and we are going to follow that.